waiter/src/waiter/shell_scheduler.clj

```diff
@@ -223,13 +248,14 @@
     (if (contains? id->service service-id)
       (let [{:keys [id->instance]} (get id->service service-id)
             {:keys [:shell-scheduler/process] :as instance} (get id->instance instance-id)]
-        (if instance
+        (if (and instance (active? instance))
```

Comment:
Why restrict to killing only active instances?

Reply:
We're in `kill-instance`, and something is only `active?` if it's not killed.

Reply:
okay